index.html

```diff
@@ -101,19 +199,623 @@ <h3>
               The port that the socket should be bound to. When omitted, or 0,
               any port can be used.
             </dd>
+            <div class="note">
+              The only current implementation (Chromium) ignores
+              {{TCPSocketOptions/localAddress}} and {{TCPSocketOptions/localPort}}.
+            </div>
+            <dt>
+              <dfn>noDelay</dfn> member
+            </dt>
+            <dd>
+              Enables the `TCP_NODELAY` option,
+              disabling <a href=
+              "https://en.wikipedia.org/wiki/Nagle%27s_algorithm">Nagle's
+              algorithm</a>.
+            </dd>
+            <dt>
+              <dfn>keepAlive</dfn> member
+            </dt>
+            <dd>
+              Enables/disables TCP Keep-Alive by setting `SO_KEEPALIVE` option on the socket. 
+            </dd>
+            <dt>
+              <dfn>keepAliveDelay</dfn> member
+            </dt>
+            <dd>
+              Specifies the Keep-Alive ping delay in milliseconds. Must be specified only if
+              {{TCPSocketOptions/keepAlive}} is set to `true`.
+            </dd>
+          </dl>
+
+        </section>
+
+      </section>
+
+      <section>
+        <h3>{{TCPSocket/[[readable]]}} attribute (internal)</h3>
+
+        <aside class="example">
+          An application receiving data from a TCP socket will typically use the {{ReadableStream}} like this:
+
+          <pre class="js">
+            const { readable } = await socket.connection;
+
+            const reader = readable.getReader();
+
+            while (true) {
+              const { value, done } = await reader.read();
+              if (done) {
+                // |reader| has been canceled.
+                break;
+              }
+              // Do something with |value|...
+            }
+
+            reader.releaseLock();
+          </pre>
+        </aside>
+      
+        The steps to <dfn data-lt="initialize tcp readable stream">initialize the TCP readable stream</dfn> are:
+
+        <ol>
+          <li>Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps [=in parallel=]:
+            <ol>
+              <li>
+                Invoke the operating system to read a tcp packet from the socket, placing the result in the [=byte sequence=] |bytes|.
+              <li>
+                [=Queue a global task=] on the [=relevant global object=] of
+                [=this=] using the [=TCP socket task source=] to run the
+                following steps:
+                <ol>
+                  <li>
+                    If EOF packet was received, run the following steps:
```

Comment:
I would say "a FIN packet was received" or "the operating system indicates the socket was closed gracefully".

Reply:
Fixed!